### PR TITLE
Multiple patches

### PR DESCRIPTION
**Pull request by r-darwish**
Opening separate pull requests is tedious, so I'll open a central one. We can discuss all patches
here, and I can individually remove changes that you do not wish to accept.

Some commits have a description message.

### CODE REVIEW

File: init.el [hunk 1]
```diff
@@ -804,6 +800,7 @@
         magit-log-arguments '("-n256" "--graph" "--decorate" "--show-signature")
         magit-log-section-arguments '("-n256" "--decorate" "--show-signature")
         magit-diff-refine-hunk t
+        magit-display-buffer-function #'magit-display-buffer-same-window-except-diff-v1
```
Comment (drrlvn) [inline init.el]: Maybe `display-buffer` is better?

File: init.el [hunk 2]
```diff
@@ -921,7 +918,7 @@ _M-p_: Unmark  _M-n_: Unmark  _q_: Quit"
          ("C-c p" . projectile-command-map))
   :config
   (projectile-register-project-type 'rust-cargo '("Cargo.toml")
-                                    :compile "cargo check --all-features --tests"
+                                    :compile "cargo check --all-features --tests --examples"
```
Comment (drrlvn) [inline init.el]: Use `cargo-process--command-check` instead of duplicating.

File: init.el [hunk 3]
```diff
@@ -1035,4 +1032,26 @@ _M-p_: Unmark  _M-n_: Unmark  _q_: Quit"
   :ensure
   :hook (js2-mode . prettier-js-mode))
 
+(use-package engine-mode
+  :ensure
+  :defer
+  :config
+  (engine-mode 1)
+
```
Comment (drrlvn) [inline init.el]: No need for so many blank lines.

File: init.el (hunk 1 above)
Comment (r-darwish) [reply init.el]: `display-buffer` seems to open the buffer in the other window

File: init.el (hunk 2 above)
Comment (r-darwish) [reply init.el]: The problem is that I need to specify `:after cargo` in `projectile` to use that variable, which means that projectile will not be loaded until cargo does.

**Comment by r-darwish** [comment]
All discussions has been resolved and I pushed one more patch

**Pull request merged**